### PR TITLE
Reject messaging promise if not embedded.

### PR DESCRIPTION
**Pull request by erwinmombay**

### CODE REVIEW

**Comment by erwinmombay** [comment]
@dvoytenko PTAL

File: src/service/viewer-impl.js [hunk 1]
```diff
@@ -276,10 +276,17 @@ export class Viewer {
...
+        new Promise((resolve, reject) => {
           /** @private @const {function(!Viewer)} */
           this.messagingReadyResolver_ = resolve;
+
+          if (!this.isEmbedded_) {
+            reject(new Error('Page is not embedded.'));
+          }
         })).catch(reason => {
+          if (!this.isEmbedded_) {
+            return;
```
Comment (dvoytenko) [inline src/service/viewer-impl.js]: This swallows the error. Why?

File: src/service/viewer-impl.js [hunk 2]
```diff
@@ -276,10 +276,17 @@ export class Viewer {
      */
     this.messagingReadyPromise_ = timer.timeoutPromise(
         20000,
-        new Promise(resolve => {
+        new Promise((resolve, reject) => {
           /** @private @const {function(!Viewer)} */
           this.messagingReadyResolver_ = resolve;
+
+          if (!this.isEmbedded_) {
+            reject(new Error('Page is not embedded.'));
```
Comment (jridgewell) [inline src/service/viewer-impl.js]: Instead of throwing an error and catching it, just `resolve()`.

File: src/service/viewer-impl.js (hunk 1 above)
Comment (cramforce) [inline src/service/viewer-impl.js]: Shouldn't we say:

if (!this.isEmbedded) {
this.messagingReadyPromise_ = Promise.reject(new Error('not meeded'))
}

and then down where we postMessage just not do it?

File: src/service/viewer-impl.js (hunk 2 above)
Comment (erwinmombay) [inline src/service/viewer-impl.js]: wasn't sure if there would be consequences down the chain for resolving it (trying to get the fastest fix in, and do a more correct one later) so a reject seemed more appropriate. will likely remove this rejection in favor of just having a conditional on the reporting.

**Comment by dvoytenko** [comment]
LGTM

**Comment by cramforce** [comment]
Merging, so we quickly get a master build.

File: src/service/viewer-impl.js (hunk 2 above)
Comment (jridgewell) [inline src/service/viewer-impl.js]: Throwing, then catching in the catch handler is the same thing with more code. :wink:

**Pull request merged**